[Cheat] Allow Elegy of Emptiness anywhere

Eblo opened the pull request:
Extremely straightforward, this cheat just circumvents the scene check when playing the song.

![image](https://github.com/HarbourMasters/2ship2harkinian/assets/7004497/16b60a0b-621c-46ed-9827-f33822a1be00)

<!--- section:artifacts:start -->
### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2038227911.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2038236160.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2038276916.zip)
<!--- section:artifacts:end -->

Eblo committed:
Add cheat to allow Elegy of Emptiness anywhere

garrettjoecox reviewed (approved):
Only point of feedback would be on the VB name, it's not horrible but still reads odd to me "Should elegy in ikana only?"

Otherwise just needs updating to modern menu and headers LGTM

Eblo committed:
Use modern menu for Elegy Anywhere, rename VB

Removed redundant null from Elegy scene check
Moved ElegyAnywhere enhancement's menu order

Eblo commented:
I applied those changes. Agreed on the VB name, so I changed it to `VB_ELEGY_CHECK_SCENE`.

I'm not sure why generating the o2r failed in the PR. It succeeded in my fork's run. I don't have the authority to clear cache or re-run the PR job here, but I'm assuming it was a fluke.

garrettjoecox commented on mm/2s2h/BenGui/SearchableMenuItems.h:
```diff
@@ -967,6 +967,12 @@ void AddEnhancements() {
+                "gCheats.ElegyAnywhere",
+                "Allows Elegy of Emptiness outside of Ikana",
+                WIDGET_CVAR_CHECKBOX,
+                {},
+                [](widgetInfo& info) { RegisterMoonJumpOnL(); } },
```
Might need to re-run prettier after this
```suggestion
                {} },
```

Eblo replied:
Whoops. Removed.

Eblo committed:
Remove extraneous widgets from ElegyAnywhere menu
